### DISCUSSION TITLE
How can I get bounding boxes (.json) from batch prediction?

### DISCUSSION THREAD

**Discussion by Wondaeng** [obss/sahi#782] [Q&A] [answered]
Hello, 
I am trying to use SAHI for batch inference in some directories with images, wrapping with it my custom function like below:

```
from sahi.predict import predict
def batch_inference(image_src_path, detection_model, project, name, slice_height = 512, slice_width = 512):
    result = predict(
        source = image_src_path,  # Path of the directory with images to be inferenced
        detection_model = detection_model,
        slice_height = slice_height,
        slice_width = slice_width,
        overlap_height_ratio = 0.2,
        overlap_width_ratio = 0.2,
        model_category_mapping = {"0":"parasite"},
        visual_export_format = 'png',
        
        # Inference images (i.e., visuals) will be saved in project/name/visuals/
        # and this export_dir will be return as return_dict parameter is set as True
        project = project,
        name = name
    )
```
It seems that I can pass something through dataset_json_path parameter to get result.json if I use coco formatted dataset.
However, my images are not a coco formatted dataset, since they are random images uploaded by users who wants to get predictions on them.
How can I get bounding boxes information too (maybe in .json) not only visuals, without using dataset_json_path parameter?

**Comment by fcakyon** [comment, marked as answer]
You can export predictions in pickle format by providing export_pickle=True

**Reply by Wondaeng** [reply]
Thank you!
